OrbitService/Utils.cpp

```diff
@@ -18,54 +21,60 @@
 #include "OrbitBase/Logging.h"
 #include "OrbitBase/Result.h"
 #include "OrbitBase/SafeStrerror.h"
-#include "Path.h"
-#include "Utils.h"
 #include "absl/strings/str_format.h"
 #include "absl/strings/strip.h"
 
-namespace LinuxUtils {
+namespace orbit_service::utils {
 
 using ::ElfUtils::ElfFile;
 using orbit_grpc_protos::ModuleInfo;
 
-outcome::result<std::vector<std::string>> ReadProcMaps(pid_t pid) {
-  std::filesystem::path maps_path{absl::StrFormat("/proc/%d/maps", pid)};
-  OUTCOME_TRY(maps_string, OrbitUtils::FileToString(maps_path));
-  return absl::StrSplit(maps_string, '\n');
+namespace {
+ErrorMessageOr<uint64_t> FileSize(const std::string& file_path) {
+  struct stat stat_buf {};
+  int ret = stat(file_path.c_str(), &stat_buf);
+  if (ret != 0) {
+    return ErrorMessage(absl::StrFormat("Unable to call stat with file \"%s\": %s", file_path,
+                                        SafeStrerror(errno)));
+  }
+  return outcome::success(stat_buf.st_size);
 }
 
-outcome::result<std::string> ExecuteCommand(const std::string& cmd) {
+ErrorMessageOr<std::string> ExecuteCommand(const std::string& cmd) {
   // TODO (antonrohr): check exit code of executed cmd. If exit code is not 0,
   //  return failure
-  std::array<char, 128> buffer;
+  std::array<char, 128> buffer{};
   std::string result;
   std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
   if (!pipe) {
     ERROR("Failed to execute command \"%s\": %s", cmd.c_str(), SafeStrerror(errno));
-    return outcome::failure(static_cast<std::errc>(errno));
+    return ErrorMessage(
+        absl::StrFormat("Failed to execute command \"%s\": %s", cmd, SafeStrerror(errno)));
   }
   while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
     result += buffer.data();
   }
   return outcome::success(result);
 }
+}  // namespace
 
 ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
```

Comment:
ReadModules maybe? (not sure about this.)

Reply:
Yes, it does not matter that much. I like ReadModules -> done